pkg/client/list_pager.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/pager"
+	// cnoclient "github.com/openshift/cluster-network-operator/pkg/client"
+)
+
+// for use with dynamic client
+func ListAllOfSpecifiedType(resourceType schema.GroupVersionResource, ctx context.Context, client Client) ([]*uns.Unstructured, error) {
+	list := []*uns.Unstructured{}
+	err := pager.New(func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
+		return client.Default().Dynamic().Resource(resourceType).List(ctx, metav1.ListOptions{})
+	}).EachListItem(ctx, metav1.ListOptions{}, func(obj runtime.Object) error {
+		list = append(list, obj.(*uns.Unstructured))
+		return nil
+	})
```

Comment:
I think doing it the current way has worked fine, and it makes accesses to items in `migration.go` more readable since it's just a slice of the items, which is all the data my code needs to be interacting with.